### PR TITLE
Bugfix: TEA2741

### PR DESCRIPTION
**Pull request by MengZhuNAV**
See: https://favro.com/organization/98c34fb974ce445eac854de0/1844bbac3b6605eacc8f5543?card=Tea-2741

In FagsakContext, the logic for updating user information was that when hentFagsak() is called, it will first change the fagsakRessurser.fagsak.status, and then call backend to get fagsak. The change of fagsakRessurser.fagsak.status will trigger another call to backend to get user information.
However, in the scenario with the bug, hentFagsak() was not called, instead, settFagsak() was called to update the fagsak, which could not trigger the call to get user information.
With this bugfix, FagsakContext updates user information when either 1) fagsak status is changed, or 2) the person ident of fagsak is no longer equal to the person ident of user information

### CODE REVIEW

File: src/frontend/context/FagsakContext.tsx [hunk 1]
```diff
@@ -33,7 +33,13 @@ const [FagsakProvider, useFagsakRessurser] = createUseContext(() => {
     const { axiosRequest } = useApp();
 
     React.useEffect(() => {
-        if (fagsakRessurser.fagsak.status === RessursStatus.SUKSESS) {
+        if (
```
Comment (henninghaakonsen) [inline src/frontend/context/FagsakContext.tsx:36]: I recommend splitting this into two useEffects. One for fetching the fagsak, and another listening on changes to fagsak.

Once the fagsak is loaded and either the person is not fetched or the fagsak is connected to another person we fetch the correct person. 

We can also replace fagsakRessurser with two separate useStates like we do with logg 👍

Comment (MengZhuNAV) [reply src/frontend/context/FagsakContext.tsx:42]: Updated👍

File: src/frontend/context/FagsakContext.tsx [hunk 2]
```diff
@@ -82,6 +73,40 @@ const [FagsakProvider, useFagsakRessurser] = createUseContext(() => {
             });
     };
 
+    const erBrukerUtdatert = (bruker: Ressurs<IPersonInfo>, fagsak: Ressurs<IFagsak>): boolean => {
```
Comment (henninghaakonsen) [inline src/frontend/context/FagsakContext.tsx:76]: Very nice! If you also could split the two states I think this is a huge improvement :)

Comment (MengZhuNAV) [reply src/frontend/context/FagsakContext.tsx:51]: Ok, done!

**Review by henninghaakonsen — APPROVED** [review]
Very good! Fyi there is a function for setting henter-state - byggHenterRessurs().

**Pull request merged**